ddl/column.go

```diff
@@ -498,17 +498,39 @@ func checkDropColumnForStatePublic(tblInfo *model.TableInfo, colInfo *model.Colu
 }
 
 func onDropColumn(t *meta.Meta, job *model.Job) (ver int64, _ error) {
-	tblInfo, colInfo, err := checkDropColumn(t, job)
+	tblInfo, colInfo, idxInfos, err := checkDropColumn(t, job)
 	if err != nil {
 		return ver, errors.Trace(err)
 	}
 
+	dependentHiddenCols := make([]*model.ColumnInfo, 0)
+	for _, indexInfo := range idxInfos {
+		for _, indexColumn := range indexInfo.Columns {
+			if tblInfo.Columns[indexColumn.Offset].Hidden {
+				dependentHiddenCols = append(dependentHiddenCols, tblInfo.Columns[indexColumn.Offset])
+			}
+		}
+	}
+
 	originalState := colInfo.State
 	switch colInfo.State {
 	case model.StatePublic:
 		// public -> write only
 		job.SchemaState = model.StateWriteOnly
 		colInfo.State = model.StateWriteOnly
+		if len(idxInfos) > 0 {
+			for _, indexInfo := range idxInfos {
+				indexInfo.State = model.StateWriteOnly
+				if len(dependentHiddenCols) > 0 {
```

Comment:
The codes here may not necessary because the `dependentHiddenCols` is used to deal with "expression index" and it is built on "virtual generated columns". We probably don't need to deal with the "dropping column with the virtual generated column" for now.

Reply:
yes, since the indexInfos comes from the concrete column, and every indexInfo only covers one column, so there is no hidden column actually.